### ISSUE TITLE
Multi-site setup issues

### ISSUE THREAD

**Issue by vandres** [nystudio107/craft-retour#241]
### Question
I am trying to get the plugin working with my multi-site setup. We have 6 different sites, with 2 different domains and 2 languages.

![image](https://user-images.githubusercontent.com/1436383/183157589-77671d45-3149-489c-81d2-4175aa94545d.png)

This is working (RegEx). 

![image](https://user-images.githubusercontent.com/1436383/183158152-3f9b375f-1f92-4299-9919-ad2c65aad97a.png)

But when I auto generate redirects "when entry urls change", it's generated like this.

![image](https://user-images.githubusercontent.com/1436383/183158273-aa0617e2-c83b-4b17-846c-a56d327a78bc.png)

This is not working, because it doesn't respect the site path.

If I change it manually to this, it is working again.

![image](https://user-images.githubusercontent.com/1436383/183158711-5ee4247e-6868-45fa-b058-f51024a264a7.png)

Am I missing something in my setup? Would it be possible, to allow configuring "exactmatch" to "regexmatch" for auto generated redirects?

**Comment by khalwat** [comment]
Probably you need to use **Entry Redirects URL Match Type** set to Full URL

**Comment by vandres** [comment]
@khalwat Thanks for the response. When creating a new rule from the 404 dashboard, neither are respected, right? Neither the mentioned setting, nor the site url. How to make this more convenient for the non-technical users?

![grafik](https://user-images.githubusercontent.com/1436383/183245781-c67b30f2-fbf9-47ad-8910-7b238baca3e1.png)

**Comment by khalwat** [comment]
Right, perhaps that should be set as the default for new redirects that are manually created too?

**Comment by vandres** [comment]
Sounds good

**Comment by vandres** [comment]
For anybody interested. I was able to resolve our setup with the following Event:

```php
        Event::on(Redirects::class, Redirects::EVENT_BEFORE_RESOLVE_REDIRECT, function (ResolveRedirectEvent $event) {
            $site = \Craft::$app->getSites()->currentSite;
            $redirects = Retour::$plugin->redirects->getAllStaticRedirects(null, $site->id);

            foreach ($redirects as $redirect) {
                if ($redirect['redirectMatchType'] !== 'exactmatch') {
                    continue;
                }
                if ($redirect['redirectSrcMatch'] !== 'pathonly') {
                    continue;
                }

                $urlToMatch = rtrim($site->baseUrl, '/') . '/' . ltrim($redirect['redirectSrcUrlParsed'], '/');
                if ($event->fullUrl !== $urlToMatch) {
                    continue;
                }

                $redirectUrl = rtrim($site->baseUrl, '/') . '/' . ltrim($redirect['redirectDestUrl'], '/');
                $event->redirectDestUrl = $redirectUrl;
                break;
            }
        });
```

**Comment by khalwat** [comment]
Ah let me re-open this so I don't miss putting the change in mentioned above.

**Comment by khalwat** [comment]
This is all much more simple now with the recent changes in Retour to account for Craft sites.
